repository/population.go

```diff
@@ -27,49 +27,46 @@ func NewPopulationRepository(mysql *sqlx.DB) PopulationRepository {
 	}
 }
 
-func (p *populationRepository) queryLog(message string, functionName string) {
-	logrus.WithFields(logrus.Fields{
+func (p *populationRepository) generateLogger(functionName string) *logrus.Entry {
+	return logrus.WithFields(logrus.Fields{
 		"Module":  "Repository",
 		"Funtion": functionName,
-	}).Info(message)
-}
-
-func (p *populationRepository) errorMessage(err error, functionName string) {
-	logrus.WithFields(logrus.Fields{
-		"Module":  "Repository",
-		"Funtion": functionName,
-	}).Error(err)
+	})
 }
 
 func (p *populationRepository) QueryAllDistrict() ([]model.District, error) {
+	logger := p.generateLogger("QueryAllDistrict")
 	res := []model.District{}
 	q := `SELECT Name, DistrictID
 	FROM District`
 	err := p.mysql.Select(&res, q)
 	if err != nil {
-		p.errorMessage(err, "QueryAllDistrict")
+		logger.Error(err)
 		return res, err
 	}
-	p.queryLog("Select all district", "QueryAllDistrict")
+	logger.Info("Select all district")
 	return res, nil
 }
+
 func (p *populationRepository) QueryTotalPopulation(districtId int) (int64, error) {
+	logger := p.generateLogger("QueryTotalPopulation")
 	var res int64
 	q := `
 	SELECT COUNT(*) as Total
 	FROM Population as p
 	WHERE DistrictId=?`
 	err := p.mysql.Get(&res, q, districtId)
 	if err != nil {
-		p.errorMessage(err, "QueryTotalPopulation")
+		logger.Error(err)
 		return res, err
 	}
-	p.queryLog(fmt.Sprintf("Get total population districtId: %d", districtId), "QueryTotalPopulation")
+	logger.Info(fmt.Sprintf("Get total population districtId: %d", districtId))
```

Comment:
No need to use fmt. You can directly input it in .Info

```suggestion
	logger.Info("Get total population districtId: ", districtId)
```

Reply:
fixed